teps/0131-tekton-conformance-policy.md

```diff
@@ -0,0 +1,251 @@
+#### 1. Do field requirements mean the same for Tekton users and vendors?
+
+The short answer is no.
+
+"REQUIRED" in the conformance policy context is, **vendors** have to support this field, while for **users** when submitting Tekton Primitives, those fields are not necessarily required. **Required fields in Tekton APIs must be all included in the required fields of Tekton Conformance APIs.**
+
+For instance, say `TaskRun.Spec.Param` is required in the **Tekton Conformance APIs**, while it is an optional field in the **Tekton APIs**:
+* Tekton **users** are **NOT** required to specify the `Param` field when defining a TaskRun.
+* Tekton **vendors MUST** support that field so that Tekton ***users*** are able to use it.
+
+**Required fields in Tekton APIs must be all included in the required fields of Tekton Conformance APIs.**
+
+If a field is required in Tekton APIs, say `taskSpec.steps`, but is not required in Conformant APIs, vendors have the freedom to not support that field.
+
+Say **Vendor A** doesn't support `taskSpec.steps` in its Conformant Tekton Pipeline Service – **Service A**. Then users are not able to reuse any Tasks that work for the community Tekton Pipelines Service to the **Service A**, because the required (from Tekton APIs perspective) `taskSpec.steps` is an unknown field to Service A.
+
+#### 2 Tekton Conformance Policy (this doc) v.s. [Tekton API Compatibility Policy][Tekton API Compatibility Policy].
+
+The two policies resemble each other in many aspects: they both define Tekton API fields, breaking changes, and  the practice of making API changes. However, in essence, they are different:
+
+The ***Conformance Policy*** defines ***Tekton Conformance API Specification*** that **vendors** MUST implement.
+* It only includes **Tekton GA Primitives**.
+* **Tekton Open Source Community** is like a "vendor" that provides the community version of Tekton "product" that contains a reference implementation that conforms to the Conformance Policy.
+
+The **Compatibility Policy** defines rules to follow when making changes on **Tekton APIs**, so that Tekton users can use it to define their CI/CD Pipeline.
+
+On the other hand, **Tekton APIs** drive the **Tekton Conformance APIs**, i.e. fields are added to the **Tekton Conformance APIs** as they graduate from **Tekton APIs**. The lifecycle of a field is:
+* Added to Tekton APIs as alpha
+* Updated to beta
+* Updated to GA (become an OPTIONAL field to Conformance APIs)
+* Added to Conformance APIs as REQUIRED
+
+Once the Conformance Policy is defined, the Tekton Compatibility Policy needs to be updated accordingly:
+
+Making backward compatible updates (in the context of Tekton Compatibility Policy)  on a Tekton API fields that's in the Conformance API (for example, adding a new field in an existing object field) is a backward incompatible change for Tekton Conformance Policy, which requires a new Conformance Policy Version (which needs to be approved by 50%+ community owners)
```

Comment:
This sentence is a bit confusing; can you reword?

Reply:
Updated, wdyt?

Reply:
How about:

If a field required for conformance is modified, even in a way that is backwards compatible for users of the API, this is a backwards incompatible change to the conformance policy. For example, ... (I actually don't really understand the example you provided so maybe there is a more concrete one?). Such a change requires a new conformance policy version and approval of >50% of owners.

Reply:
A good example is adding a new [ResultType](https://github.com/tektoncd/pipeline/blob/58712bbdf5a03228e6d304ad6a8ae0457171c810/pkg/apis/pipeline/v1/result_types.go#L62-L66)(let's assume this field is required just for the explanation purpose), which is an additive change for Tekton APIs (compatible), but new Vendors have to support the new type to claim the conformance.

Reply:
thanks, that's helpful! if the new result type is optional but all the other result types are required, is that still a backwards incompatible change to conformance?

Reply:
And actually we are trying to only define required fields in the Conformance Policy, that's said, all fields that are not included in the Conformance Policy are optional by default.